docs/dev/table/sqlClient.md

```diff
@@ -0,0 +1,538 @@
+{% endhighlight %}
+
+You can use the following query to see both result modes in action:
+
+{% highlight sql %}
+SELECT name, COUNT(*) AS cnt FROM (VALUES ('Bob'), ('Alice'), ('Greg'), ('Bob')) AS NameTable(name) GROUP BY name 
+{% endhighlight %}
+
+This query performs a bounded word count example. The following sections explain how to read from table sources and configure other table program properties. 
+
+{% top %}
+
+Configuration
+-------------
+
+The SQL Client can be started with the following optional CLI commands. They are discussed in detail in the subsequent paragraphs.
+
+{% highlight text %}
+./bin/sql-client.sh embedded --help
+
+Mode "embedded" submits Flink jobs from the local machine.
+
+  Syntax: embedded [OPTIONS]
+  "embedded" mode options:
+     -d,--defaults <environment file>      The environment properties with which
+                                           every new session is initialized.
+                                           Properties might be overwritten by
+                                           session properties.
+     -e,--environment <environment file>   The environment properties to be
+                                           imported into the session. It might
+                                           overwrite default environment
+                                           properties.
+     -h,--help                             Show the help message with
+                                           descriptions of all options.
+     -j,--jar <JAR file>                   A JAR file to be imported into the
+                                           session. The file might contain
+                                           user-defined classes needed for the
+                                           execution of statements such as
+                                           functions, table sources, or sinks.
+                                           Can be used multiple times.
+     -l,--library <JAR directory>          A JAR file directory with which every
+                                           new session is initialized. The files
+                                           might contain user-defined classes
+                                           needed for the execution of
+                                           statements such as functions, table
+                                           sources, or sinks. Can be used
+                                           multiple times.
+     -s,--session <session identifier>     The identifier for a session.
+                                           'default' is the default identifier.
+{% endhighlight %}
+
+{% top %}
+
+### Environment Files
+
+A SQL query needs a configuration environment in which it is executed. The so-called *environment files* define available table sources and sinks, external catalogs, user-defined functions, and other properties required for execution and deployment.
+
```

Comment:
There is typo in the link "`http://http://yaml.org/`". Also I would go for sth like:

Every environment file is a regular [YAML file](http://yaml.org/). An example of such file is presented below. This configuration:
 - defines an environment with a table source `MyTableName` that reads from CSV file,
 - specifies a parallelism of 1 for queries executed in this environment, 
 - specifies an even-time characteristic, and
 - runs queries in the `table` result mode.
 
Just because it is easy for the eye to focus on what the script specifies.